app/facades/bill_search_facade.rb

```diff
@@ -2,25 +2,66 @@ class BillSearchFacade
   def initialize(params)
     @params = params
   end
-  
+
   def bills
     bill_search = BillService.new(@params).bills_by_query
     bill_objects = []
-    bill_search[:searchresult].drop(1).map do |bill_data|
+    bill_search[:searchresult].drop(1).each do |bill_data|
       bill_hash = {id: bill_data.first.to_s.to_i,
         attributes: bill_data.last }
-        
-        bill_objects << ApiBill.new(bill_hash)
-      end
+        bill_object = ApiBill.new(bill_hash)
+        bill_objects << bill_object
       
+      create_update_bills(bill_data.last)
+    end
       bill_objects
+  end
+
+  def bill
+    bill_search = BillService.new(@params).bills_by_id
+    bill_hash = { id: bill_search[:bill][:bill_id],
+      attributes: bill_search[:bill]
+    }
+    bill_object = ApiBill.new(bill_hash)
+    bill_object
+  end
```

Comment:
we should be able to remove the variable being called here "bill_object" since Api.new(bill_hash) is the last thing being called. so it could be:
def bill
    bill_search = BillService.new(@params).bills_by_id
    bill_hash = { id: bill_search[:bill][:bill_id],
      attributes: bill_search[:bill]
    }
    bill_object = ApiBill.new(bill_hash)
 end
 
 
 just a small thing. all good if you want to leave it